framework/src/outputs/BlockRestrictionDebugOutput.C

```diff
@@ -0,0 +1,309 @@
+  const auto & material_warehouse = _problem_ptr->getMaterialWarehouse();
+
+  // get all user objects
+  std::vector<UserObject *> userObjects;
+  _problem_ptr->theWarehouse()
+      .query()
+      .condition<AttribSystem>("UserObject")
+      .condition<AttribThread>(0)
+      .queryIntoUnsorted(userObjects);
+
+  // do we have to check all object types?
+  const bool include_all = !_scope.isValid() || _scope.contains("all");
+
+  // iterate all subdomains
+  for (const auto & subdomain_id : mesh_subdomains)
+  {
+    // get the corresponding subdomain name
+    const auto & subdomain_name = mesh.getSubdomainName(subdomain_id);
+
+    out << "\n";
+    out << "  Subdomain '" << subdomain_name << "' (id " << subdomain_id << "):\n";
+
+    bool objectsFound = false;
+
+    // Variables
+    if (include_all || _scope.contains("variables"))
+    {
+      std::set<std::string> names;
+      for (unsigned int var_num = 0; var_num < _sys.n_vars(); var_num++)
+      {
+        const auto & var_name = _sys.variable_name(var_num);
+        if (_problem_ptr->hasVariable(var_name))
+        {
+          const MooseVariableBase & var =
+              _problem_ptr->getVariable(/*tid = */ 0,
+                                        var_name,
+                                        Moose::VarKindType::VAR_ANY,
+                                        Moose::VarFieldType::VAR_FIELD_ANY);
+          if (var.hasBlocks(subdomain_id))
+            names.insert(var_name);
+        }
+      }
+      objectsFound = printCategoryAndNames("Variables", names) || objectsFound;
+    }
+
+    // Kernels
+    if (include_all || _scope.contains("kernels"))
+    {
+      std::set<std::string> names;
+      for (const auto & kernel : kernels)
+      {
+        if (kernel->hasBlocks(subdomain_id))
+          names.insert(kernel->name());
+      }
+      objectsFound = printCategoryAndNames("Kernels", names) || objectsFound;
+    }
+
+    // AuxVariables
+    if (include_all || _scope.contains("auxvariables"))
+    {
+      std::set<std::string> names;
+      const auto & sys = auxSystem.system();
+      for (unsigned int vg = 0; vg < sys.n_variable_groups(); vg++)
+      {
+        const VariableGroup & vg_description(sys.variable_group(vg));
+        for (unsigned int vn = 0; vn < vg_description.n_variables(); vn++)
+        {
+          if (vg_description.active_on_subdomain(subdomain_id))
+            names.insert(vg_description.name(vn));
+        }
+      }
+      objectsFound = printCategoryAndNames("AuxVariables", names) || objectsFound;
+    }
+
+    // AuxKernels
+    if (include_all || _scope.contains("auxkernels"))
+    {
+
+      {
+        const auto & wh = auxSystem.nodalAuxWarehouse();
+        std::set<std::string> names;
+        if (wh.hasActiveBlockObjects(subdomain_id))
+        {
+          const auto & auxkernels = wh.getActiveBlockObjects(subdomain_id);
+          for (auto & auxkernel : auxkernels)
+            names.insert(auxkernel.get()->name());
```

Comment:
```suggestion
            names.insert(auxkernel->name());
```
same for all the others